core/beacon/src/types.rs

```diff
@@ -48,15 +48,27 @@ impl BeaconBlock {
                 parent_hash,
                 merkle_root_tx: MerkleHash::default(),
                 merkle_root_state,
-                signature: primitives::signature::default_signature(),
+                signature: DumbGroupSignature::default(),
                 index,
             },
             transactions,
         }
     }
 
-    pub fn sign(&mut self, signer: &Arc<Signer>) {
-        self.header.signature = signer.sign(&self.hash());
+    // Since the signature is contained in the header, we need a hash that omits it
+    fn hash_for_signing(&self) -> CryptoHash {
+        // TODO: find a better way to do this
+        let dummy_block = BeaconBlockHeader{signature: DumbGroupSignature::default(), ..self.header};
+        dummy_block.hash()
+    }
+
+    pub fn sign(&mut self, signer: &dyn Signer) {
+        let sig = signer.sign(&self.hash_for_signing());
+        self.header.signature.add(signer.public_key(), sig);
+    }
+
+    pub fn verify_signature(&self) -> bool {
```

Comment:
I put a placeholder `validate_signature` in node/client/src/lib.rs. You might want to remove that if you decide to have the verification here.